crypto/cmp/cmp_client.c

```diff
@@ -106,9 +106,12 @@ static int save_statusInfo(OSSL_CMP_CTX *ctx, OSSL_CMP_PKISI *si)
     ss = si->statusString; /* may be NULL */
     for (i = 0; i < sk_ASN1_UTF8STRING_num(ss); i++) {
         ASN1_UTF8STRING *str = sk_ASN1_UTF8STRING_value(ss, i);
+        ASN1_UTF8STRING *dup = ASN1_STRING_dup(str);
 
-        if (!sk_ASN1_UTF8STRING_push(ctx->statusString, ASN1_STRING_dup(str)))
+        if (!dup || !sk_ASN1_UTF8STRING_push(ctx->statusString, dup)) {
+            ASN1_UTF8STRING_free(dup);
```

Comment:
Won't this only fix a memory leak the first time in the loop? If we've already pushed one string, come round again, and the second `sk_ASN1_UTF8STRING_push()` fails, we won't clear up the first duplicated string or push.

It doesn't look like callers of this function do any clearing up if this function returns `0`.

Reply:
It should be cleared when OSSL_CMP_CTX_free() is called on the ctx.

Reply:
Ah, okay, thanks